rts/lra/client/pom.xml

```diff
@@ -29,6 +29,16 @@
       <artifactId>lra-service-base</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
```

Comment:
Why do we need these and where is the evidence that they are required. Is there an example of something failing, and if not can we have a reproducer.

Reply:
Basically, when the LRA client is employed, an implementation of the JAX-RS Client API must be specified among the dependencies. As we already specify that we want to use `org.jboss.resteasy.microprofile.client.impl.MpClientBuilderImpl` in the ServiceLoader, I think it might be beneficial to also add the resteasy-client dependency.

For what concerns `resteasy-jackson2-provider`, when the client is used to fetch all LRAs (getAllLRAs()), an exception RESTEASY003145 is thrown.

Reproducer:
```
/**
 * Copyright Red Hat
 *
 * SPDX-License-Identifier: LGPL-2.1-only
 */

import io.narayana.lra.LRAData;
import io.narayana.lra.client.NarayanaLRAClient;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

public class CheckDeploymentTestCase {

    // I suppose you have a running lra-coordinator at this address
    private final NarayanaLRAClient lraClient =
            new NarayanaLRAClient("http://127.0.0.1:8080/lra-coordinator/lra-coordinator");

    @Test
    public void testWildflyIsRunning() {
        final List<LRAData> allLRAs = lraClient.getAllLRAs();

        assertNotNull(allLRAs);
        assertTrue(allLRAs.isEmpty());
    }
}
```

Reply:
Can you add this as a unit test.

Reply:
I have added this TestCase in the (future) PR for [JBTM-3470](https://issues.redhat.com/browse/JBTM-3470). If you also want to test lra-client on its own, I can add a bit of code to test it with Arquillian :-)

Reply:
Your test case would not need to have any server running, I think just asserting that `java.lang.ClassNotFoundException` is not thrown when calling `lraClient.getAllLRAs()` would be enough (or asserting 404 or whatever gets returned from the call when there is no server).

Reply:
I think that a server (basically, an Arquillian container) should be used for testing the client as more unit tests should be written to test different aspects of lra-client (for example, the exception thrown when `resteasy-jackson2-provider` is removed)

Reply:
What about creating a new JIRA to track unit tests? I would like to keep this PR focused on dependencies. @mmusgrov If you agree with this, could you please resolve this conversation?

Reply:
It's not normal to add functionality and to then defer the test to a future patch.
But I guess in this instance it's not a big enough deal to argue the case so I'll approve the PR and you can merge it.